test/document.test.js

```diff
@@ -14225,6 +14225,47 @@ describe('document', function() {
     assert.strictEqual(duplicateKeyError.message, 'Email must be unique');
     assert.strictEqual(duplicateKeyError.cause.code, 11000);
   });
+
+  it('supports global transforms per schematype (gh-15084)', async function () {
+    class SchemaCustomType extends mongoose.SchemaType {
+      static schemaName = 'CustomType';
```

Comment:
It looks like eslint needs a new option set: [`ecmaVersion: 2022`](https://stackoverflow.com/questions/60046847/eslint-does-not-allow-static-class-properties), this should be possible as the [target node (16.20), supports it](https://node.green/#ES2022).
Or alternatively use the old syntax to define statics.

Reply:
In the interim I removed the offending code in [08113ef](https://github.com/Automattic/mongoose/pull/15163/commits/08113efa08ccec25265523a92a8075a48aabf057)